src/animal/animal.controller.ts

```diff
@@ -21,27 +22,37 @@ import {
   ApiResponse,
   ApiTags,
 } from '@nestjs/swagger';
-import { AnimalService } from './animal.service';
-import { fromAnimalModelToDto, fromAnimalMutateDtoToModel, fromAnimalsModelToDtos } from '../mappers/animal.mapper';
-import { AnimalDto } from 'src/dto/read.animal.dto';
-import { IncreaseAgeDto, IncreaseWeightDto, MutateAnimalDto } from 'src/dto/mutate.animal.dto';
-import { ANIMALS_API, APPLICATION_JSON_MEDIA_TYPE } from 'src/utils/constants';
+import { ANIMALS_API, APPLICATION_JSON_MEDIA_TYPE } from 'src/shared/constants';
+import {
+  fromAnimalsModelToDtos,
+  fromAnimalModelToDto,
+  fromAnimalMutateDtoToModel,
+} from 'src/shared/mapper/animal.mapper';
+import { AnimalDto, IncreaseAgeDto, IncreaseWeightDto } from 'src/shared/dto/animal/animal.dto';
+import { AnimalCreateDto } from 'src/shared/dto/animal/create-animal-dto';
+import { IAnimalService } from './interface/animal.service.inteface';
+import { ANIMAL_SERVICE_KEY } from './animal.providers';
 
 @ApiBearerAuth()
 @ApiTags(ANIMALS_API)
 @Controller(ANIMALS_API)
 export class AnimalController {
-  constructor(private service: AnimalService) {}
+  constructor(@Inject(ANIMAL_SERVICE_KEY) private service: IAnimalService) {}
 
   @Get()
+  @ApiOperation({ summary: 'Get all animals' })
   @ApiResponse({ description: 'Retrieved list of animals successfully.', type: AnimalDto, isArray: true })
+  @ApiProduces(APPLICATION_JSON_MEDIA_TYPE)
   async getAllAnimal(): Promise<AnimalDto[]> {
```

Comment:
Don't use `then` if you don't need it. 

```suggestion
  async getAllAnimal(): Promise<AnimalDto[]> {
  const animals = await this.service.getAll();
  const animalDtos = fromAnimalsModelToDtos(animals)
  
   return animalDtos
  }
```